### PR TITLE
Add CODEOWNERS

### PR DESCRIPTION
**Pull request by TanayParikh**
Modelled after https://github.com/dotnet/aspnetcore/blob/main/.github/CODEOWNERS.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+*                                                                                 @Pilchie
+/global.json                                                                      @dotnet/aspnet-build
+/.vscode/                                                                         @dotnet/aspnet-build @TanayParikh
+/.devcontainer/                                                                   @dotnet/aspnet-build @TanayParikh
+/.github/                                                                         @dotnet/aspnet-build
+/src/                                                                             @dotnet/aspnet-blazor-eng
+/eng/                                                                             @dotnet/aspnet-build
+/eng/common/                                                                      @dotnet-maestro-bot
+/eng/Versions.props                                                               @dotnet-maestro-bot @dougbu
```
Comment (wtgodbe) [inline .github/CODEOWNERS:12]: Change dougbu to aspnet-build for this

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+/eng/Version.Details.xml                                                          @dotnet-maestro-bot @dougbu
```
Comment (wtgodbe) [inline .github/CODEOWNERS:13]: and this

File: .github/CODEOWNERS (hunk 1 above)
Comment (TanayParikh) [reply .github/CODEOWNERS]: Just removed `@dougbu` given `/eng/` is already `dotnet/aspnet-build`

**Review by wtgodbe — APPROVED** [review]

**Review by pranavkm — APPROVED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (dougbu) [reply .github/CODEOWNERS]: @TanayParikh you should probably follow up w/ a fix here. The eng/common/ rule supersedes everything mentioned for eng/

Comment (TanayParikh) [reply .github/CODEOWNERS]: https://github.com/dotnet/razor-compiler/pull/39

**Pull request merged**